src/lib.rs

```diff
@@ -105,11 +105,26 @@ pub async fn create_docker_image(
     plan_options: &GeneratePlanOptions,
     build_options: &DockerBuilderOptions,
 ) -> Result<()> {
-    let app = App::new(path)?;
+    let mut app = App::new(path)?;
     let environment = Environment::from_envs(envs)?;
 
     let mut generator = NixpacksBuildPlanGenerator::new(get_providers(), plan_options.clone());
-    let plan = generator.generate_plan(&app, &environment)?;
+    let planned = generator.generate_plan(&app, &environment)?;
+    let plan = planned.0;
+    app = planned.1;
+
+    let input_path = std::fs::canonicalize(path)?;
+    let len = input_path.display().to_string().len();
+    let sub = &app.paths.get(0).unwrap().display().to_string()[len..];
+    if sub.split(std::path::MAIN_SEPARATOR).count() > 1 {
+        let mut path = sub.split(std::path::MAIN_SEPARATOR).collect::<Vec<&str>>();
+        path.remove(path.len() - 1);
+        path.remove(0);
+        println!(
+            "Using subdirectory \"{}\"",
+            path.join(std::path::MAIN_SEPARATOR.to_string().as_str())
+        );
+    }
```

Comment:
This can be cleaned up a bit by using `Path::strip_prefix`

```rs
  let app = App::new(path)?;
    let environment = Environment::from_envs(envs)?;

    let orig_path = app.source.clone();

    let mut generator = NixpacksBuildPlanGenerator::new(get_providers(), plan_options.clone());
    let (plan, app) = generator.generate_plan(&app, &environment)?;

    if let Ok(subdir) = app.source.strip_prefix(orig_path) {
        if subdir != std::path::Path::new("") {
            println!(
                "Using subdirectory \"{}\"",
                subdir.to_str().unwrap().to_string()
            );
        }
    }
```

Reply:
this doesnt seem to work correctly